analyses/cell-types-annotation/run-merge-cell-types-annotations-all.R

```diff
@@ -0,0 +1,284 @@
+      
+      seurat_obj@meta.data <- seurat_obj@meta.data %>%
+        rename_with(~ paste("broad_", ., sep = ""), matches("^scores\\."))
+      
+      # We want to attach the metadata related to the cell type annotation from other methods to the ones of the object
+      # We randomly chose the object from the first method, `SingleR_broad`
+      
+      # Read metadata `fine_SingleR`
+      new_metadata <- readr::read_tsv(fine_SingleR_file, guess_max = 100000, show_col_types = FALSE) %>%
+        mutate(pruned.labels.fine = pruned.labels) %>%
+        
+        # Select for columns to use for join to the object
+        select(pruned.labels.fine, singler.fine, matches("^scores\\.")) %>% # start with
+        rename_with(~ paste("fine_", ., sep = ""), matches("^scores\\."))
+      
+      # Check if the number of rows in new_metadata matches the number of cells in the Seurat object
+      if (nrow(new_metadata) == ncol(seurat_obj)) {
+        # Add metadata to Seurat object
+        seurat_obj <- AddMetaData(seurat_obj, metadata = new_metadata)
+
+      } else {
+        # If the number of rows doesn't match, print a warning
+        warning("Number of rows in new_metadata doesn't match the number of cells in seurat_obj. Label metadata will not be added to the Seurat object.")
+      }
+    
+    } else if (method == "singler&gene_markers"){
+      
+      ## Input files
+      broad_SingleR_file <- file.path(broad_SingleR_results_dir, "seurat_obj_SingleR_broad.rds")
+      fine_SingleR_file <- file.path(fine_SingleR_results_dir, "metadata.tsv")
+      gene_markers_file <- file.path(gene_markers_results_dir, "metadata.tsv")
+
+      # Read seurat object #########################
+      ##############################################
+      seurat_obj <- readRDS(broad_SingleR_file)
+      
+      # Rename the column in the metadata of the Seurat object
+      seurat_obj@meta.data$pruned.labels.broad <- seurat_obj@meta.data$pruned.labels
+      
+      seurat_obj@meta.data <- seurat_obj@meta.data %>%
+        rename_with(~ paste("broad_", ., sep = ""), matches("^scores\\."))
+      
+      # We want to attach the metadata related to the cell type annotation from other methods to the ones of the object
+      # We randomly chose the object from the first method, `SingleR_broad`
+      
+      # Read metadata `fine_SingleR`
+      fine_SingleR_df <- readr::read_tsv(fine_SingleR_file, guess_max = 100000, show_col_types = FALSE) %>%
+        mutate(pruned.labels.fine = pruned.labels) %>%
+        
+        # Select for columns to use for join to the object
+        select(cell, pruned.labels.fine, singler.fine, matches("^scores\\.")) %>% # start with
+        rename_with(~ paste("fine_", ., sep = ""), matches("^scores\\."))
+      
+      
+      # Read metadata `gene_markers`
+      gene_markers_df <- readr::read_tsv(gene_markers_file, guess_max = 100000, show_col_types = FALSE) %>%
```

Comment:
The `gene_markers` metadata should be named as `new_metadata` when it is read in. Otherwise, `new_metadata` won't be defined for line `182` and will lead to an error (see error below).


```
Error: object 'new_metadata' not found
Execution halted
```

```suggestion
      # Read metadata `gene_markers`
      new_metadata <- readr::read_tsv(gene_markers_file, guess_max = 100000, show_col_types = FALSE) %>%
```